### PR TITLE
creates a content change detail component

### PR DESCRIPTION
**Pull request by akash1810**
smaller components make things easier to reason about

### CODE REVIEW

File: public/video-ui/src/components/ContentChangeDetails/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  render() {
+    const { video } = this.props;
+
+    if (!video || !video.id) {
+      return null;
+    }
+
+    return (
+      <ManagedForm>
+        {this.getCreatedByField(video)}
```
Comment (mbarton) [inline public/video-ui/src/components/ContentChangeDetails/index.js]: The functions don't have any parameters in their definitions?

File: public/video-ui/src/components/ContentChangeDetails/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      >
+        <TextInput />
+      </ManagedField>
+    );
+  };
+
+  render() {
+    const { video } = this.props;
+
+    if (!video || !video.id) {
```
Comment (mbarton) [inline public/video-ui/src/components/ContentChangeDetails/index.js]: `video` is marked as required in the props. Would we ever get a video without an id?

File: public/video-ui/src/components/ManagedForm/ManagedForm.js [hunk 3]
```diff
@@ -55,7 +57,7 @@ export class ManagedForm extends React.Component {
         ManagedForm.managedTypes.indexOf(child.type) > -1
           ? React.cloneElement(child, {
               data: this.props.data,
-              updateData: this.props.updateData,
+              updateData: this.props.updateData || this.emptyUpdateData,
```
Comment (mbarton) [inline public/video-ui/src/components/ManagedForm/ManagedForm.js]: What's the reason for the change here?

Comment (akash1810) [reply public/video-ui/src/components/ManagedForm/ManagedForm.js]: there isn't one!

File: public/video-ui/src/components/ContentChangeDetails/index.js (hunk 1 above)
Comment (akash1810) [reply public/video-ui/src/components/ContentChangeDetails/index.js]: ah! great spot, thanks!

File: public/video-ui/src/components/ContentChangeDetails/index.js (hunk 2 above)
Comment (akash1810) [reply public/video-ui/src/components/ContentChangeDetails/index.js]: This is for atom creation, as the component is shared. An incoming PR will tidy this up a bit, pushing this further up the tree for a cleaner and clearer design.

File: public/video-ui/src/components/ManagedForm/ManagedForm.js (hunk 3 above)
Comment (akash1810) [reply public/video-ui/src/components/ManagedForm/ManagedForm.js]: removed.

File: public/video-ui/src/components/ContentChangeDetails/index.js (hunk 1 above)
Comment (akash1810) [reply public/video-ui/src/components/ContentChangeDetails/index.js]: fixed.

File: public/video-ui/src/components/ManagedForm/ManagedForm.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { ManagedSection } from './ManagedSection';
 export class ManagedForm extends React.Component {
   static propTypes = {
     data: PropTypes.object,
-    updateData: PropTypes.func.isRequired,
+    updateData: PropTypes.func,
```
Comment (mbarton) [inline public/video-ui/src/components/ManagedForm/ManagedForm.js]: Is this still left over?

Comment (akash1810) [reply public/video-ui/src/components/ManagedForm/ManagedForm.js]: Should have added a comment here, sorry.

This is a necessary change. We're using `ManagedForm` for the CCD component, however the CCD component is completely read-only, so doesn't need to specify an `updateData` function... even if it did, it would be `() => undefined` which is pointless. Much nicer to not make `updateData` required.

Comment (akash1810) [reply public/video-ui/src/components/ManagedForm/ManagedForm.js]: Obviously we could not use `ManagedForm` for this component, but I cba to work out the styling atm 😄 .
